algolia.js

```diff
@@ -93,11 +88,25 @@ async function index() {
 
   const raw = fs.readFileSync('dist/pages.json', 'utf-8');
   /** @type {AlgoliaItem[]} */
-  const algoliaData = JSON.parse(raw).map((e) => {
-    // Set date of when object is being added to algolia.
-    e.indexedOn = indexedOn.getTime();
-    return e;
-  });
+  const algoliaData = JSON.parse(raw)
+    .map((/** @type {AlgoliaItem} */ e) => {
+      addLocaleToUrl(e.locale, e.url);
+      // Set date of when object is being added to algolia.
+      e.indexedOn = indexedOn.getTime();
+      return e;
+    })
+    .map((/** @type {AlgoliaItem} */ e) => {
```

Comment:
Optional: Having 2 map functions operating on the same array seems usually suspicious (raises question of "why double loop?"). You can improve readability by adding a comment, or by splitting it into 2 statements:

const posts = JSON.parse(raw);
const urlToLocale = posts.reduce(...);   // Here you build up your locale dict (equivalent of 1st loop)
const algoliaData = posts.map(...);   // (2nd loop)

This also lets you avoid having names like "urlToLocale" and "addLocaleToUrl" which are confusing what you add to whom.